examples/blinky/CMakeLists.txt

```
@@ -4,9 +4,25 @@ set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/../../cmake/stm32_gcc.cmake
 project(stm32-blinky C ASM)
 set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
 
-find_package(CMSIS COMPONENTS STM32L0 STM32F1 STM32F4 REQUIRED)
-find_package(HAL COMPONENTS STM32L0 STM32F1 STM32F4 RCC GPIO CORTEX REQUIRED)
-# Find all drivers:
+# F4 example enabled by default
+option(F4_EXAMPLE "Compile F4 example" ON)
+option(F1_EXAMPLE "Compile F1 example" OFF)
+option(L0_EXAMPLE "Compile L0 example" OFF)
+
+find_package(CMSIS COMPONENTS STM32F4 REQUIRED)
+find_package(HAL COMPONENTS STM32F4 RCC GPIO CORTEX REQUIRED)
+
+if(L0_EXAMPLE)
+    find_package(CMSIS COMPONENTS STM32L0 REQUIRED)
+    find_package(HAL COMPONENTS STM32L0 REQUIRED)
```

I  didn't test but shouldn't it be this ?
```suggestion
    find_package(HAL COMPONENTS STM32L0 RCC GPIO CORTEX REQUIRED)
```

---

selecting no component will look up all components, so this should work.

---

But is it what we want ? and it's not same as L4 lines.
If we add a comment it's OK for me

---

Updated to use `find_package` with one call by filling component lists